romancal/dq_init/dq_init_step.py

```diff
@@ -44,24 +43,25 @@ def process(self, input):
         # Convert to RampModel if needed
         if not isinstance(input_model, RampModel):
             # Create base ramp node with dummy values (for validation)
-            input_ramp = maker_utils.mk_ramp(shape=input_model.shape)
+            output_model = RampModel.make_default(shape=input_model.shape)
 
             # Copy input_model contents into RampModel
-            for key in input_model.keys():
+            for key, value in input_model:
                 # If a dictionary (like meta), overwrite entires (but keep
                 # required dummy entries that may not be in input_model)
-                if isinstance(input_ramp[key], dict):
-                    input_ramp[key].update(input_model.__getattr__(key))
-                elif isinstance(input_ramp[key], np.ndarray):
+                if key in output_model and isinstance(output_model[key], dict):
+                    output_model[key].update(value)
+                elif key in output_model and isinstance(output_model[key], np.ndarray):
                     # Cast input ndarray as RampModel dtype
-                    input_ramp[key] = input_model.__getattr__(key).astype(
-                        input_ramp[key].dtype
-                    )
+                    output_model[key] = value.astype(output_model[key].dtype)
                 else:
-                    input_ramp[key] = input_model.__getattr__(key)
+                    output_model[key] = value
+
+            # Remove the "resultantdq" attribute as this is accidentally moved and breaks
+            # the regression tests
+            if "resultantdq" in output_model.model_extra:
+                del output_model.model_extra["resultantdq"]
```

Comment:
We would make a separate ticket for this one?

Reply:
Probably